src/jdk.compiler/share/classes/com/sun/tools/javac/code/Type.java

```diff
@@ -1320,36 +1338,76 @@ public boolean isReference() {
 
         @Override
         public boolean isPrimitiveClass() {
```

Comment:
Agreed, per earlier comment, I will take up this in JDK-8268734. 

(I am a bit unsettled on using "value type" at all, since the JEP introduces the name primitive class type and we are moving away value/inline nomenclature. But I see that java.lang.Class defines a method by name asValueType)